packages/clarity-js/src/data/upload.ts

```diff
@@ -151,8 +151,14 @@ function send(payload: string, zipped: Uint8Array, sequence: number, beacon: boo
         // However, we don't want to rely on it for every payload, since we have no ability to retry if the upload failed.
         // Also, in case of sendBeacon, we do not have a way to alter HTTP headers and therefore can't send compressed payload
         if (beacon && "sendBeacon" in navigator) {
-            dispatched = navigator.sendBeacon(url, payload);
-            if (dispatched) { done(sequence); }
+            try {
+                // sendBeacon must be binded to the navigator before passing the reference
+                dispatched = navigator.sendBeacon.bind(navigator)(url, payload);
```

Comment:
I looked up issue you referenced #194 - do you know if this is a Sentry problem? Looked up MDN: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon and it doesn't require binding to navigator before sending sendBeacon. If you could explain more that will help.

Reply:
@sarveshnagpal I couldn't find more details also on the manner, but this fix is used in popular frameworks like next.js, you can check this [commit](https://github.com/blitz-js/next.js/commit/4bb419b473394285249044c174a8ba736669f98f) from the next.js repo , they had the same issue and fixed like I did.

Reply:
@sarveshnagpal here is the [issue](https://github.com/vercel/next.js/issues/23856) that mansions this in the next.js repo

Reply:
sentry also had the same issue and fixed it like that in this [PR ](https://github.com/getsentry/sentry-javascript/pull/4008)